pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4095,9 +4093,6 @@ cdef int get_day_of_month(datetime other, day_opt) except? -1:
     elif day_opt == 'business_end':
         # last business day of month
         return get_lastbday(other.year, other.month)
-    elif is_integer_object(day_opt):
-        days_in_month = get_days_in_month(other.year, other.month)
-        return min(day_opt, days_in_month)
     elif day_opt is None:
         # Note: unlike `shift_month`, get_day_of_month does not
         # allow day_opt = None
```

Comment:
I would check if we have tests for this raise case (also doesn't need to be an else anymore as everything returns. followon on).

Reply:
yes, we have tests for both the ValueError and NotImplementedError cases here

Reply:
as for "else", I get compile-time failures about unreached code